sigopt/ratelimit.py

```diff
@@ -0,0 +1,27 @@
+import random
+import threading
+import time
+
+
+global_failed_status_count = 0
+thread_lock = threading.Lock()
```

Comment:
It'll be easier to test if you make these object properties and just have a global instance of `FailedStatusRateLimit`

Reply:
yeah, dunno why i assumed i should lock just the increment. Updated.